### PR TITLE
feat(ChipGroup): Add event handler prop for overflow-chip click 

### LINKED ISSUE #5190: Provide event for click on collapsedText or expandedText button in ChipGroup.
**Describe the issue. What is the expected and unexpected behavior?**

Currently, there is no event is fired when user click on collapsedText or expandedText in ChipGroup.
We should provide handler to the parent component. It will helps to perform some operation on the click of collapsedText or expandedText button.

**Please provide the steps to reproduce. Feel free to link CodeSandbox or another tool.**
Enhancement

**Is this a bug or enhancement? If this issue is a bug, is this issue blocking you or is there a work-around?**
Enhancement
**What is your product and what release version are you targeting?**

### PR DESCRIPTION
**Pull request by joeldmyers**
**What**: Closes #5190 

Adds an optional prop called `onOverflowChipClick` that, when used, calls the provided handler function when the overflow (toggle expand/collapse) chip is clicked.  

There is already an onClick prop, hence naming this differently.  Let me know if you have a better idea for a name for this prop.

//cc @tlabaj

### CODE REVIEW

File: packages/react-core/src/components/ChipGroup/ChipGroup.tsx [hunk 1]
```diff
@@ -156,7 +160,14 @@ export class ChipGroup extends React.Component<ChipGroupProps, ChipGroupState> {
               ))}
               {numChildren > numChips && (
                 <li className={css(styles.chipGroupListItem)}>
-                  <Chip isOverflowChip onClick={this.toggleCollapse} component="button">
+                  <Chip
+                    isOverflowChip
+                    onClick={event => {
+                      this.toggleCollapse();
+                      onOverflowChipClick(event);
```
Comment (tlabaj) [inline packages/react-core/src/components/ChipGroup/ChipGroup.tsx:167]: Shouldn't calling this be  this be conditional on wether it as an overflow chip?

**Review by tlabaj — CHANGES_REQUESTED** [review]
Could you also please update the react-integration demo app and add the new prop to verify the API typing.  And can you also add a integration test.

File: packages/react-core/src/components/ChipGroup/ChipGroup.tsx (hunk 1 above)
Comment (joeldmyers) [reply packages/react-core/src/components/ChipGroup/ChipGroup.tsx:167]: From what I understand, line 156 in ChipGroup.tsx is generating all the non-overflow chips passed through as children, and then this last Chip is by definition always the overflow chip This chip is only rendered if numChildren > numChips (i.e., if we have overflow).  Here, there is the `isOverflowChip` which is shorthand for `isOverflowChip={true}`. 

So this code here should be only applicable to the overflow chip.

**Review by nicolethoen — COMMENTED** [review]
Once the prop is included in relevant tests, it looks good :)

**Comment by joeldmyers** [comment]
OK I added a demo component with this, as well as an integration test.  Hopefully I understood everything you wanted correctly; please lmk if any other changes are needed.  Thanks!

**Review by tlabaj — APPROVED** [review]
Lgtm

**Review by nicolethoen — APPROVED** [review]

**Review by tlabaj — APPROVED** [review]

**Pull request merged**